### PR TITLE
[stable/jaeger-operator] Update to 1.14.0

### PR DESCRIPTION
**Pull request by gmaiztegi**
Signed-off-by: Gorka Maiztegi <gmaiztegi@reviewpro.com>

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [ ] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/jaeger-operator/Chart.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
 description: jaeger-operator Helm chart for Kubernetes
 name: jaeger-operator
-version: 2.9.0
-appVersion: 1.13.1
+version: 2.10.0
+appVersion: 1.14.0
```
Comment (davidkarlsen) [inline stable/jaeger-operator/Chart.yaml:5]: can you update the values in the readme to match? TIA

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/jaeger-operator/Chart.yaml (hunk 1 above)
Comment (gmaiztegi) [reply stable/jaeger-operator/Chart.yaml:5]: Absolutely. Sorry for not doing that earlier.

**Comment by shamil** [comment]
I see https://github.com/helm/charts/pull/18110 already merged, but it doesn't work with 1.14
this PR has the right fix https://github.com/helm/charts/pull/18139/files#diff-37c3fcded6f4093dbb64c75182318d32R47

**Comment by davidkarlsen** [comment]
@shamil #18110 is another chart...

**Review by davidkarlsen — APPROVED** [review]

**Comment by shamil** [comment]
Oops, I meant https://github.com/helm/charts/pull/18216

**Comment by qlikcoe** [comment]
I'm sorry, I didn't know this PR also exists. 

Follow up PR for fixing a `POD_NAMESPACE` bug and resources update. 
https://github.com/helm/charts/pull/18320

**Comment by zanhsieh** [comment]
@gmaiztegi 
Could you bump the chart version to latest and fix the conflicting files please?
stable/jaeger-operator/Chart.yaml
stable/jaeger-operator/templates/crd.yaml
stable/jaeger-operator/templates/role.yaml

**Comment by gmaiztegi** [comment]
The last changes by @shamil already include my changes, so I'm closing this PR.

**Pull request closed without merging**